lib/radar_client.js

```diff
@@ -1,65 +1,51 @@
 /* globals setImmediate */
-var log = require('minilog')('radar_client'),
-    MicroEE = require('microee'),
+var MicroEE = require('microee'),
     eio = require('engine.io-client'),
     Scope = require('./scope.js'),
     StateMachine = require('./state.js'),
     immediate = typeof setImmediate != 'undefined' ? setImmediate :
                                     function(fn) { setTimeout(fn, 1); };
 
+MicroEE.mixin(Client);
+
 function Client(backend) {
   var self = this;
+  this.logger = require('minilog')('radar_client');
   this._ackCounter = 1;
   this._channelSyncTimes = {};
-  this._users = {};
+  this._uses = {};
   this._presences = {};
   this._subscriptions = {};
   this._restoreRequired = false;
   this._queuedMessages = [];
   this._isConfigured = false;
 
-  // Allow backend substitution for tests
-  this.backend = backend || eio;
-
   this._createManager();
   this.configure(false);
+  this._addListeners();
 
-  this.on('authenticateMessage', function(message) {
-    if (this._configuration) {
-      message.userData = this._configuration.userData;
-      if (this._configuration.auth) {
-        message.auth = this._configuration.auth;
-        message.userId = this._configuration.userId;
-        message.userType = this._configuration.userType;
-        message.accountName = this._configuration.accountName;
-      }
-    }
-    this.emit('messageAuthenticated', message);
-  });
-
-  this.on('messageAuthenticated', function(message) {
-    this._sendMessage(message);
-  });
+  // Allow backend substitution for tests
+  this.backend = backend || eio;
 }
 
-MicroEE.mixin(Client);
```

Comment:
Works fine either way - before or after.  I sometimes find this 30 lines down into the file, so I prefer it in the same place in all files.